Makefile

```
@@ -62,41 +91,57 @@ _api.clean.rpcmanager:
 api/rpcmanager.proto: go.sum go.mod
 	go run github.com/gfanton/grpcutil/rpcmanager/protofile > $@
 src/api/rpcmanager.pb.js: api/rpcmanager.proto
-	$(pbjs) -t json-module -w es6 -o $@ $<
+	$(pbjs) -t json-module -w commonjs -o $@ $<
 src/api/rpcmanager.pb.d.ts: api/rpcmanager.proto
 	$(pbjs) -t static-module $< | $(pbts) -o $@ -
 
 .PHONY: api.generate _api.generate.rpcmanager _api.generate.protocol
 
-## go bind
+# - Bind : Handle gomobile bind
 
-bind.init: $(TMPDIR)/.tool-versions .cache/bind/gomobile
-.cache/bind/gomobile: go.sum go.mod
+# - Bind - initialization
+bind_init_files := $(TMPDIR)/.tool-versions $(gobind) $(gomobile)
+
+$(gobind): go.sum go.mod
+	@mkdir -p $(dir $@)
+	go build -o $@ golang.org/x/mobile/cmd/gobind && chmod +x $@
+
+$(gomobile): $(gobind) go.sum go.mod
 	@mkdir -p $(dir $@)
 	go build -o $@ golang.org/x/mobile/cmd/gomobile && chmod +x $@
-	$@ init
-# FIXME(gfanton): find a more elegant way to make asdf works in the tmp directory
+	$(gomobile) init || (rm -f $@ && exit 1) # in case of failure, remove gomobile so we can init again
+
 $(TMPDIR)/.tool-versions: .tool-versions
-	@echo "> copying current `.tool-versions` in '$(TMPDIR)' folder in order to make asdf works"
+	@echo "> copying current '.tool-versions' in '$(TMPDIR)' folder in order to make asdf works"
 	@echo "> this hack is needed in order for gomobile (who is building from '$(TMPDIR)') bind to use the correct javac and go version"
 	@cp -v $< $@
-bind.clean: _bind.clean.framework
-	rm -f $(TMPDIR)/.tool-versions
-	rm -rf .cache/bind
 
-# use `nowatchdog` tags to build, see https://github.com/libp2p/go-libp2p-connmgr/issues/98
-go_files := $(shell find . -iname '*.go')
-go_deps := go.mod go.sum $(go_files)
-$(OUTPUT_FRAMEWORK): bind.init $(go_deps)
+.PHONY: bind.init
+
+# - Bind - ios framework
+
+$(weshnetcore_xcframework): $(bind_init_files) $(go_deps)
 	@mkdir -p $(dir $@)
-	.cache/bind/gomobile bind -v \
+    # need to use `nowatchdog` tags, see https://github.com/libp2p/go-libp2p-connmgr/issues/98
+	$(gomobile) bind -v \
 		-tags 'nowatchdog' -prefix=Weshnet \
-		-o $@ -target ios $(make_dir)/framework/core
-_bind.clean.framework:
-	rm -rf $(OUTPUT_FRAMEWORK)
+		-o $@ -target ios ./framework/core
+_bind.clean.ios:
+	rm -rf weshnetcore_xcframework
 
-output/android/core.aar: bind.init $(go_deps)
-	@mkdir -p $(dir $@)
-	.cache/bind/gomobile bind -v -o $@ -target android $(make_dir)/framework/core
+# - Bind - android aar and jar
+
+$(weshnetcore_jar): $(ANDROID_OUTPUT_LIBS_DIR)/WeshnetCore.aar
+$(weshnetcore_aar): $(bind_init_files) $(go_deps)
+	@mkdir -p $(dir $@) .cache/bind/android
+	$(gomobile) bind -v \
+		-javapkg=network.weshnet \
+		-o $@ -target android ./framework/core
```

```suggestion
		-o $@ -target android -androidapi 21 ./framework/core
```

I have the following issue if I don't set the `androidapi` vesion:

```
go build -o /Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile golang.org/x/mobile/cmd/gomobile && chmod +x /Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile
/Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile init || (rm -f /Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile && exit 1) # in case of failure, remove gomobile so we can init again
/Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile bind -v \
                -javapkg=network.weshnet \
                -o /Users/xxx/dev/berty/expo/weshnet-expo/android/libs/WeshnetCore.aar -target android ./framework/core
/Users/xxx/dev/berty/expo/weshnet-expo/.cache/bind/gomobile: ANDROID_NDK_HOME specifies /Users/xxx/Library/Android/sdk/ndk/25.2.9519653, which is unusable: unsupported API version 16 (not in 19..33)
make: *** [/Users/xxx/dev/berty/expo/weshnet-expo/android/libs/WeshnetCore.aar] Error 1
```